src/Cli/dotnet/commands/dotnet-workload/install/WorkloadManifestUpdater.cs

```diff
@@ -104,6 +112,41 @@ public async Task BackgroundUpdateAdvertisingManifestsWhenRequiredAsync()
             }
         }
 
+        private void WriteUpdatableWorkloadsFile()
+        {
+            var installedWorkloads = _workloadRecordRepo.GetInstalledWorkloads(_sdkFeatureBand);
+            var updatableWorkloads = GetUpdatableWorkloadsToAdvertise(installedWorkloads);
+            var filePath = GetAdvertisingWorkloadsFilePath();
+            var jsonContent = JsonSerializer.Serialize(updatableWorkloads.Select(workload => workload.ToString()).ToArray());
+            if (Directory.Exists(Path.GetDirectoryName(filePath)))
+            {
+                Directory.CreateDirectory(Path.GetDirectoryName(filePath));
+            }
+            File.WriteAllText(filePath, jsonContent);
+        }
+
+        public static void AdvertiseWorkloadUpdates()
+        {
+            try
+            {
+                var backgroundUpdatesDisabled = bool.TryParse(Environment.GetEnvironmentVariable("DOTNET_CLI_WORKLOAD_UPDATE_NOTIFY_DISABLE"), out var disableEnvVar) && disableEnvVar;
+                var adUpdatesFile = Path.Combine(CliFolderPathCalculator.DotnetHomePath, ".dotnet", ".workloadAdvertisingUpdates");
```

Comment:
Should this use the `GetAdvertisingWorkloadsFilePath` method?  Should it be factored out into a static method so the logic can be shared?